### ISSUE TITLE
Job scheduler

### ISSUE THREAD

**Issue by z3n4j** [Myddleware/myddleware#281]
Hi. If I execute manually active rule the task edn correctly.

If the task run with cron job scheduler i get this error:

Failed to create documents : Notice: Undefined index: values /var/www/html/src/Myddleware/RegleBundle/Classes/rule.php Line : ( 305 )Failed to create documents : Notice: Undefined index: values /var/www/html/src/Myddleware/RegleBundle/Classes/rule.php Line : ( 305 )

**Comment by Myddleware** [comment]
Do you have the same php version with your browser and with your ssh command ? So could you check with the command `php -v` in a ssh tool like putty ? 
Could you also check that the user which runs the command has the permission to write int the cache/log directory. You can run this command : `php bin/symfony_requirements`

**Comment by z3n4j** [comment]
PHP 7.2.28 (cli) (built: Feb 18 2020 11:07:04) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.28, Copyright (c) 1999-2018, by Zend Technologies

**Comment by z3n4j** [comment]
Symfony Requirements Checker
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> PHP is using the following php.ini file:
  /etc/php.ini

> Checking Symfony requirements:
  ...............................W........


 [OK]
 Your system is ready to run Symfony projects


Optional recommendations to improve your setup
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 * intl ICU version installed on your system is outdated (62.1) and
   does not match the ICU data bundled with Symfony (65.1)
   > To get the latest internationalization data upgrade the ICU
   > system package and the intl PHP extension.


Note  The command console could use a different php.ini file
~~~~  than the one used with your web server. To be on the
      safe side, please check the requirements from your web
      server using the web/config.php script.

**Comment by Myddleware** [comment]
Could you try to run your rule with this command : 
`php bin/console myddleware:synchro <your rule id> --env=background`
Do you have the same error ?

**Comment by z3n4j** [comment]
php bin/console myddleware:synchro terpin_shop --env=background
1;5e6a0739f01972.59182377
terpin_shop
Read data for rule : terpin_shop
terpin_shop : Create documents.
terpin_shop : Number of documents created : 7
0;Failed to create documents : Notice: Undefined index: values /var/www/html/src/Myddleware/RegleBundle/Classes/rule                                       .php Line : ( 305 )

**Comment by Myddleware** [comment]
Ok so you get the same error. 
It seems that Myddleware read 7 record in your source application but has no values for them. It should never happen. 
So we will have to debug to see what the read function returns. 
Could you tell me what is your source application ? 
COuld you also add the code `print_r($result);` just before the return. If it is MySQL, add this in the file <myddleware_root>\src\Myddleware\RegleBundle\Solutions\database.php : 
![image](https://user-images.githubusercontent.com/11330301/76510186-7e298400-6451-11ea-83d2-ece1fa5044b1.png)

Then run your rule again. Please could you post what you get ?
Thanks.

**Comment by z3n4j** [comment]
Source applicatino is MsSql. 
I have to add the code in the file <myddleware_root>\src\Myddleware\RegleBundle\Solutions\database.php : or in another file?

**Comment by z3n4j** [comment]
If i execute all active task, I get no error.
![Screen Shot 03-12-20 at 11 15 AM](https://user-images.githubusercontent.com/18057413/76511268-3b68ab80-6453-11ea-90e0-673a74bb9550.PNG)

**Comment by z3n4j** [comment]
![Screen Shot 03-12-20 at 11 19 AM](https://user-images.githubusercontent.com/18057413/76511290-43c0e680-6453-11ea-9b74-8f7bd481fd9c.PNG)

**Comment by Myddleware** [comment]
Yes it is the right file. Please run your rule with your command : 
`php bin/console myddleware:synchro terpin_shop --env=background`

You should see the array $result, I would like to see it. Thanks.

**Comment by z3n4j** [comment]
[janez@localhost html]$ php bin/console myddleware:synchro terpin_shop --env=background
1;5e6a0fb0486b05.08561698
terpin_shop
Read data for rule : terpin_shop
terpin_shop : Create documents.
Array
(
)
terpin_shop : Number of documents created : 0
[janez@localhost html]$ php bin/console myddleware:synchro terpin_shop_opis_slo --env=background                               1;5e6a1009e9fc94.96194066
terpin_shop_opis_slo
Read data for rule : terpin_shop_opis_slo
terpin_shop_opis_slo : Create documents.
Array
(
)
terpin_shop_opis_slo : Number of documents created : 0
[janez@localhost html]$

**Comment by Myddleware** [comment]
OK there is no more data to read in your table. So could you please change your reference date in your rule, You can go back a few seconds in the past: 
![image](https://user-images.githubusercontent.com/11330301/76513065-001bac00-6456-11ea-93b1-370c67700771.png)
Then run again your rule, Myddleware will have something to read. 

Could you also add this "echo 'query : '.$requestSQL.chr(10);" in your database file : 
![image](https://user-images.githubusercontent.com/11330301/76513302-64d70680-6456-11ea-8eb0-f66917ec5f1f.png)

It will display the query sent to your database. It could be usefull to see what's going on.

**Comment by z3n4j** [comment]
[janez@localhost html]$ php bin/console myddleware:synchro terpin_shop --env=background
1;5e6a1517a69962.74888250
terpin_shop
Read data for rule : terpin_shop
terpin_shop : Create documents.
query : SELECT [Prezzo], [Flag_VetrinaWeb], [Flag_ArticoliWeb], [Profondita], [DescrizioneEstesaIT], [Codice], [Barcode], [Altezza], [CodiceIVA], [DescrRegFiscali], [Descrizione], [UnitaMisura], [Peso], [Larghezza], [Data_Modifica] FROM [WEB_ARTICOLI_PRESTASHOP] WHERE [Data_Modifica] > '2020-03-12 08:17:00' ORDER BY [Data_Modifica] ASC OFFSET 0 ROWS FETCH NEXT 101 ROWS ONLY
Array
(
    [count] => 7
    [error] => Error : Notice: Undefined index: deletion /var/www/html/src/Myddleware/RegleBundle/Solutions/database.php Line : ( 418 )
)
terpin_shop : Number of documents created : 7
0;Failed to create documents : Notice: Undefined index: values /var/www/html/src/Myddleware/RegleBundle/Classes/rule.php Line : ( 305 )

**Comment by Myddleware** [comment]
Thank you. We have fixed the bug here : https://github.com/Myddleware/myddleware/commit/990d92f61a1f634825f2a01fb1a36e5e6b16b3c5
You can undo all your modification in the file database.php
We 'll release the hotfix version today. We 'll keep you posted. Then you could upgrade Myddleware.

**Comment by Myddleware** [comment]
We have merged the hotfix branch : #282 
You can upgrade Myddleware in version 2.5.3d and try again.
